### PR TITLE
[Keyboard] Add a new keyboard Chidori.

### PR DESCRIPTION
**Pull request by ka2hiro**
## Description
Add a new keyboard Chidori to qmk.
Yet another split keyboard made with only through-hole components.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by ka2hiro** [comment]
@noroadsleft 
Hello, Is there anything I can do to get this PR reviewed?

**Comment by noroadsleft** [comment]
@ka2hiro Sorry, I labeled it but forgot to tag anyone.

Split keyboards aren't my area of expertise at all, but I'll review what I can.

File: keyboards/chidori/chidori.h [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        { L07, L08, L09, L10, L11, L12 }, \
+        { L13, L14, L15, L16, L17, L18 }, \
+        { L19, L20, L21, L22, L23, L24 }, \
+        { R01, R02, R03, R04, R05, R06 }, \
+        { R07, R08, R09, R10, R11, R12 }, \
+        { R13, R14, R15, R16, R17, R18 }, \
+        { R19, R20, R21, R22, R23, R24 }  \
+    }
+
+#define LAYOUT3( \
```
Comment (noroadsleft) [inline keyboards/chidori/chidori.h]:
```suggestion
#define LAYOUT_extended( \
```

File: keyboards/chidori/keymaps/extended/keymap.c [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+#include "board.h"
+
+extern keymap_config_t keymap_config;
```
Comment (noroadsleft) [inline keyboards/chidori/keymaps/extended/keymap.c]:
```suggestion
```

File: keyboards/chidori/keymaps/extended/keymap.c [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+   * |------+------+------+------+------+------|             |------+------+------+------+------+------|             |------+------+------|  +   |------+------|
+   * | Ctrl |   A  |   S  |   D  |   F  |   G  |             |   H  |   J  |   K  |   L  |   ;  |  '   |             |   4  |   5  |   6  |      |   /  | Tab  |
+   * |------+------+------+------+------+------|             |------+------+------+------+------+------|             |------+------+------+------+------+------|
+   * | Shift|   Z  |   X  |   C  |   V  |   B  |             |   N  |   M  |   ,  |   .  |   /  |Enter |             |   1  |   2  |   3  |      |   *  |  =   |
+   * |------+------+------+------+------+------|             |------+------+------+------+------+------|             |-------------+------|Enter |------+------|
+   * |ADJUST| Esc  | Alt  | GUI  |LOWER |Space |             | Space| RAISE| Left | Down |  Up  | Right|             |      0      |   .  |      |   -  | Bksp |
+   * `-----------------------------------------'             `-----------------------------------------'             `-----------------------------------------'
+   */
+    [_QWERTY] = LAYOUT3(
```
Comment (noroadsleft) [inline keyboards/chidori/keymaps/extended/keymap.c]:
```suggestion
    [_QWERTY] = LAYOUT_extended(
```

File: keyboards/chidori/keymaps/extended/keymap.c [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+   * |------+------+------+------+------+------|             |------+------+------+------+------+------|             |------+------+------|  +   |------+------|
+   * | Ctrl |   A  |   R  |   S  |   T  |   D  |             |   H  |   N  |   E  |   I  |   O  |  '   |             |   4  |   5  |   6  |      |   /  | Tab  |
+   * |------+------+------+------+------+------|             |------+------+------+------+------+------|             |------+------+------+------+------+------|
+   * | Shift|   Z  |   X  |   C  |   V  |   B  |             |   K  |   M  |   ,  |   .  |   /  |Enter |             |   1  |   2  |   3  |      |   *  |  =   |
+   * |------+------+------+------+------+------|             |------+------+------+------+------+------|             |-------------+------|Enter |------+------|
+   * |ADJUST| Esc  | Alt  | GUI  |LOWER |Space |             | Space| RAISE| Left | Down |  Up  | Right|             |      0      |   .  |      |   -  | Bksp |
+   * `-----------------------------------------'             `-----------------------------------------'             `-----------------------------------------'
+   */
+    [_COLEMAK] = LAYOUT3(
```
Comment (noroadsleft) [inline keyboards/chidori/keymaps/extended/keymap.c]:
```suggestion
    [_COLEMAK] = LAYOUT_extended(
```

File: keyboards/chidori/keymaps/extended/keymap.c [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+   * |------+------+------+------+------+------|             |------+------+------+------+------+------|             |------+------+------|  +   |------+------|
+   * | Ctrl |   A  |   O  |   E  |   U  |   I  |             |   D  |   H  |   T  |   N  |   S  |  /   |             |   4  |   5  |   6  |      |   /  | Tab  |
+   * |------+------+------+------+------+------|             |------+------+------+------+------+------|             |------+------+------+------+------+------|
+   * | Shift|   ;  |   Q  |   J  |   K  |   X  |             |   B  |   M  |   W  |   V  |   Z  |Enter |             |   1  |   2  |   3  |      |   *  |  =   |
+   * |------+------+------+------+------+------|             |------+------+------+------+------+------|             |-------------+------|Enter |------+------|
+   * |ADJUST| Esc  | Alt  | GUI  |LOWER |Space |             | Space| RAISE| Left | Down |  Up  | Right|             |      0      |   .  |      |   -  | Bksp |
+   * `-----------------------------------------'             `-----------------------------------------'             `-----------------------------------------'
+   */
+    [_DVORAK] = LAYOUT3(
```
Comment (noroadsleft) [inline keyboards/chidori/keymaps/extended/keymap.c]:
```suggestion
    [_DVORAK] = LAYOUT_extended(
```

File: keyboards/chidori/keymaps/extended/keymap.c [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+   * |------+------+------+------+------+------|             |------+------+------+------+------+------|             |------+------+------|  +   |------+------|
+   * |      |      |      |      |      |      |             |   -  |   _  |   +  |   {  |   }  |  |   |             |   4  |   5  |   6  |      |   /  | Tab  |
+   * |------+------+------+------+------+------|             |------+------+------+------+------+------|             |------+------+------+------+------+------|
+   * |      |      |      |      |      |      |             |      |      |      | Home | End  |      |             |   1  |   2  |   3  |      |   *  |  =   |
+   * |------+------+------+------+------+------|             |------+------+------+------+------+------|             |-------------+------|Enter |------+------|
+   * |      |      |      |      |      |      |             |      |      | Next | Vol- | Vol+ | Play |             |      0      |   .  |      |   -  | Bksp |
+   * `-----------------------------------------'             `-----------------------------------------'             `-----------------------------------------'
+   */
+    [_LOWER] = LAYOUT3(
```
Comment (noroadsleft) [inline keyboards/chidori/keymaps/extended/keymap.c]:
```suggestion
    [_LOWER] = LAYOUT_extended(
```

File: keyboards/chidori/keymaps/extended/keymap.c [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+   * |------+------+------+------+------+------|             |------+------+------+------+------+------|             |------+------+------|  +   |------+------|
+   * |      |  F1  |  F2  |  F3  |  F4  |  F5  |             |  F6  |   -  |   =  |   [  |   ]  |  \   |             |   4  |   5  |   6  |      |   /  | Tab  |
+   * |------+------+------+------+------+------|             |------+------+------+------+------+------|             |------+------+------+------+------+------|
+   * |      |  F7  |  F8  |  F9  |  F10 |  F11 |             |  F12 |      |      |      |      |                    |   1  |   2  |   3  |      |   *  |  =   |
+   * |------+------+------+------+------+------|             |------+------+------+------+------+------|             |-------------+------|Enter |------+------|
+   * |      |      |      |      |      |      |             |      |      | Next | Vol- | Vol+ | Play |             |      0      |   .  |      |   -  | Bksp |
+   * `-----------------------------------------'             `-----------------------------------------'             `-----------------------------------------'
+   */
+    [_RAISE] = LAYOUT3(
```
Comment (noroadsleft) [inline keyboards/chidori/keymaps/extended/keymap.c]:
```suggestion
    [_RAISE] = LAYOUT_extended(
```

File: keyboards/chidori/keymaps/extended/keymap.c [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+   * |------+------+------+------+------+------|             |------+------+------+------+------+------|             |------+------+------|  +   |------+------|
+   * | Caps |      |      |      |      | Mac  |             | Win  |  -   |   =  |Print |ScLock|Pause |             |   4  |   5  |   6  |      |   /  | Tab  |
+   * |------+------+------+------+------+------|             |------+------+------+------+------+------|             |------+------+------+------+------+------|
+   * |      |      |      |      |      |      |             |      |      |      |      |      |      |             |   1  |   2  |   3  |      |   *  |  =   |
+   * |------+------+------+------+------+------|             |------+------+------+------+------+------|             |-------------+------|Enter |------+------|
+   * |      |      |      |      |      |      |             |      |      | Home |PageDn|PageUp| End  |             |      0      |   .  |      |   -  | Bksp |
+   * `-----------------------------------------'             `-----------------------------------------'             `-----------------------------------------'
+   */
+    [_ADJUST] = LAYOUT3(
```
Comment (noroadsleft) [inline keyboards/chidori/keymaps/extended/keymap.c]:
```suggestion
    [_ADJUST] = LAYOUT_extended(
```

File: keyboards/chidori/keymaps/extended/keymap.c [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+                update_tri_layer(_LOWER, _RAISE, _ADJUST);
+            }
+            return false;
+        case ADJUST:
+            if (record->event.pressed) {
+                layer_on(_ADJUST);
+            } else {
+                layer_off(_ADJUST);
+            }
+            return false;
```
Comment (noroadsleft) [inline keyboards/chidori/keymaps/extended/keymap.c]: Same as on default keymap.

File: keyboards/chidori/readme.md [hunk 10]
```diff
@@ -0,0 +1,16 @@
+# chidori
+
+![Chidori](https://imgur.com/QvpLiOv)
```
Comment (noroadsleft) [inline keyboards/chidori/readme.md]: This needs to be a direct image link.

```suggestion
![Chidori](https://i.imgur.com/QvpLiOvl.jpg)
```

File: keyboards/chidori/readme.md [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+
+Yet another split keyboard made with only through-hole components.
+
+Keyboard Maintainer: [ka2hiro](https://github.com/ka2hiro) [@ka2hiro](https://twitter.com/ka2hiro)
```
Comment (noroadsleft) [inline keyboards/chidori/readme.md]: We changed this section to be a list a while back.

```suggestion
* Keyboard Maintainer: [ka2hiro](https://github.com/ka2hiro) [@ka2hiro](https://twitter.com/ka2hiro)
```

File: keyboards/chidori/readme.md [hunk 12]
```diff
@@ -0,0 +1,16 @@
+# chidori
+
+![Chidori](https://imgur.com/QvpLiOv)
+
+Yet another split keyboard made with only through-hole components.
+
+Keyboard Maintainer: [ka2hiro](https://github.com/ka2hiro) [@ka2hiro](https://twitter.com/ka2hiro)  
+Hardware Supported: Chidori PCB, ATMEGA328P
```
Comment (noroadsleft) [inline keyboards/chidori/readme.md]:
```suggestion
* Hardware Supported: Chidori PCB, ATMEGA328P
```

File: keyboards/chidori/readme.md [hunk 13]
```diff
@@ -0,0 +1,16 @@
+# chidori
+
+![Chidori](https://imgur.com/QvpLiOv)
+
+Yet another split keyboard made with only through-hole components.
+
+Keyboard Maintainer: [ka2hiro](https://github.com/ka2hiro) [@ka2hiro](https://twitter.com/ka2hiro)  
+Hardware Supported: Chidori PCB, ATMEGA328P  
+Hardware Availability: [@kagizaraya](https://twitter.com/kagizaraya)
```
Comment (noroadsleft) [inline keyboards/chidori/readme.md]:
```suggestion
* Hardware Availability: [@kagizaraya](https://twitter.com/kagizaraya)
```

File: keyboards/chidori/rules.mk [hunk 14]
```diff
@@ -0,0 +1,58 @@
+SRC += matrix.c
+SRC += board.c
+SRC += i2c_master.c
+
```
Comment (noroadsleft) [inline keyboards/chidori/rules.mk]: Lines 1-4 should be deleted. (We'll add them back, but modified, later in the file.)

File: keyboards/chidori/rules.mk [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+# MCU name
+MCU = atmega328p
+PROTOCOL = VUSB
+F_CPU = 16000000
```
Comment (noroadsleft) [inline keyboards/chidori/rules.mk]: Lines 7-8 can be deleted.

File: keyboards/chidori/rules.mk [hunk 16]
```diff
@@ -0,0 +1,58 @@
...
+
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   atmega32a    bootloadHID
```
Comment (noroadsleft) [inline keyboards/chidori/rules.mk]:
```suggestion
#   ATmega32A    bootloadHID
#   ATmega328P   USBasp
```

File: keyboards/chidori/rules.mk [hunk 17]
```diff
@@ -0,0 +1,58 @@
...
+# BOOTLOADER = atmel-dfu
```
Comment (noroadsleft) [inline keyboards/chidori/rules.mk]:
```suggestion
```

File: keyboards/chidori/rules.mk [hunk 18]
```diff
@@ -0,0 +1,58 @@
...
+BOOTLOADER = USBasp
+#PROGRAM_CMD = avrdude -c usbasp -p m328p -v -U flash:w:$(BUILD_DIR)/$(TARGET).hex
```
Comment (noroadsleft) [inline keyboards/chidori/rules.mk]:
```suggestion
```

File: keyboards/chidori/rules.mk [hunk 19]
```diff
@@ -0,0 +1,58 @@
...
+# If you don't know the bootloader type, then you can specify the
+# Boot Section Size in *bytes* by uncommenting out the OPT_DEFS line
+#   Teensy halfKay      512
+#   Teensy++ halfKay    1024
+#   Atmel DFU loader    4096
+#   LUFA bootloader     4096
+#   USBaspLoader        2048
+# OPT_DEFS += -DBOOTLOADER_SIZE=4096
+
+
```
Comment (noroadsleft) [inline keyboards/chidori/rules.mk]: Lines 25-35 should be deleted.

File: keyboards/chidori/rules.mk [hunk 20]
```diff
@@ -0,0 +1,58 @@
...
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration(+1000)
```
Comment (noroadsleft) [inline keyboards/chidori/rules.mk]: We're working on deleting these notes from the repository.

```suggestion
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
```

File: keyboards/chidori/rules.mk [hunk 21]
```diff
@@ -0,0 +1,58 @@
...
+MOUSEKEY_ENABLE = no        # Mouse keys(+4700)
```
Comment (noroadsleft) [inline keyboards/chidori/rules.mk]:
```suggestion
MOUSEKEY_ENABLE = no        # Mouse keys
```

File: keyboards/chidori/rules.mk [hunk 22]
```diff
@@ -0,0 +1,58 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control(+450)
```
Comment (noroadsleft) [inline keyboards/chidori/rules.mk]:
```suggestion
EXTRAKEY_ENABLE = yes       # Audio control and System control
```

File: keyboards/chidori/rules.mk [hunk 23]
```diff
@@ -0,0 +1,58 @@
...
+CONSOLE_ENABLE = no         # Console for debug(+400)
```
Comment (noroadsleft) [inline keyboards/chidori/rules.mk]:
```suggestion
CONSOLE_ENABLE = no         # Console for debug
```

File: keyboards/chidori/rules.mk [hunk 24]
```diff
@@ -0,0 +1,58 @@
...
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support (+2400 to 4200, depending on config)
```
Comment (noroadsleft) [inline keyboards/chidori/rules.mk]:
```suggestion
MIDI_ENABLE = no            # MIDI support
```

File: keyboards/chidori/rules.mk [hunk 25]
```diff
@@ -0,0 +1,58 @@
...
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs (+400)
```
Comment (noroadsleft) [inline keyboards/chidori/rules.mk]:
```suggestion
HD44780_ENABLE = no        # Enable support for HD44780 based LCDs
```

File: keyboards/chidori/rules.mk [hunk 26]
```diff
@@ -0,0 +1,58 @@
...
+
+CUSTOM_MATRIX = yes
+
```
Comment (noroadsleft) [inline keyboards/chidori/rules.mk]: `i2c_master.c` needs to be added to `QUANTUM_LIB_SRC` so the build doesn't break if Link Time Optimization is enabled.

```suggestion
# project specific files
SRC += matrix.c
SRC += board.c
QUANTUM_LIB_SRC += i2c_master.c
```

File: keyboards/chidori/chidori.c [hunk 27]
```diff
@@ -0,0 +1,51 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "chidori.h"
+
+// Optional override functions below.
```
[the remark this answers is not in the document]
Comment (ka2hiro) [reply keyboards/chidori/chidori.c]: Deleted unnecessary lines.

File: keyboards/chidori/keymaps/default/keymap.c [hunk 28]
```diff
@@ -0,0 +1,201 @@
...
+                update_tri_layer(_LOWER, _RAISE, _ADJUST);
+            }
+            return false;
+        case ADJUST:
+            if (record->event.pressed) {
+                layer_on(_ADJUST);
+            } else {
+                layer_off(_ADJUST);
+            }
+            return false;
```
[the remark this answers is not in the document]
Comment (ka2hiro) [reply keyboards/chidori/keymaps/default/keymap.c]: Changed layer keys handling and deleted unnecessary lines.

File: keyboards/chidori/keymaps/extended/keymap.c (hunk 9 above)
Comment (ka2hiro) [reply keyboards/chidori/keymaps/extended/keymap.c]: Made the same changes as the default keymap.

File: keyboards/chidori/rules.mk (hunk 14 above)
Comment (ka2hiro) [reply keyboards/chidori/rules.mk]: Modified these lines and moved to the bottom of the file.

File: keyboards/chidori/rules.mk (hunk 15 above)
Comment (ka2hiro) [reply keyboards/chidori/rules.mk]: Deleted these lines.

File: keyboards/chidori/rules.mk (hunk 19 above)
Comment (ka2hiro) [reply keyboards/chidori/rules.mk]: Deleted these lines.

File: keyboards/chidori/rules.mk (hunk 26 above)
Comment (ka2hiro) [reply keyboards/chidori/rules.mk]: Fixed it.

**Comment by ka2hiro** [comment]
@noroadsleft 
Thank you for your prompt response!
I fixed all you pointed out.

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**